source/mainnet/smart-contracts/guides/setup-contract.rst

```diff
@@ -89,6 +89,31 @@ The crate documentation is on docs.rs_.
       [dependencies]
       concordium-std = { path = "./path/to/concordium-std" }
 
+.. _setup-wee-alloc-feature:
+
+.. note::
+
+   To compile your smart contracts, a memory `allocator <https://docs.rs/concordium-std/6.0.0/concordium_std/#use-a-custom-allocator>`_ is used.
+   ``concordium-std`` version ``<6.0.0`` hard-coded the use of the `wee_alloc <https://docs.rs/wee_alloc/>`_ allocator.
+   In ``concordium-std`` version ``>=6.0.0``, ``wee_alloc`` is a feature and needs to be explicitly enabled.
+   When ``std`` feature is enabled the allocator provided by the Rust standard library is used
+   by default but when the ``wee_alloc`` feature is enabled in addition, ``wee_alloc`` is used instead.
+
+   You can enable the ``wee_alloc`` feature in ``concordium-std`` version ``>=6.0.0`` by using:
+
+   .. code-block:: rust
+
+      [features]
+      default = ["std", "wee_alloc"]
+      std = ["concordium-std/std"]
+      wee_alloc = ["concordium-std/wee_alloc"]
+
+   The main reason for using ``wee_alloc`` instead of the default allocator
+   is that ``wee_alloc`` has a smaller code footprint, i.e, the resulting smart contracts modules are going to
```

Comment:
Can you also state the main reason for why it is optional in the first place, namely that iti s unmaintained?